pkg/jobrunaggregator/jobrunbigqueryloader/uploader.go

```diff
@@ -48,46 +59,64 @@ func (o *allJobsLoaderOptions) Run(ctx context.Context) error {
 
 	logrus.Infof("Launching threads to upload test runs for %d jobs", jobCount)
 
-	waitGroup := sync.WaitGroup{}
-	errCh := make(chan error, len(jobs))
+	wg := sync.WaitGroup{}
+	jobChan := make(chan jobrunaggregatorapi.JobRow, jobCount)
+	errChan := make(chan error, jobCount)
+	// Max concurrent job imports but note there's another layer of concurrency beneath this for the file uploads
+	for i := 0; i < 20; i++ {
+		wg.Add(1)
+		go o.processJob(ctx, i, jobCount, jobChan, errChan)
+	}
+
 	for i := range jobs {
 		job := jobs[i]
+		jobChan <- job
+	}
+
+	wg.Wait()
+	close(jobChan)
```

Comment:
you need to close the channel before you wait for the jobChan consumer to be finished, right?

Reply:
Fixed.